clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/images/memory/MemoryAccessObject.kt

```diff
@@ -0,0 +1,101 @@
+package com.clevertap.android.sdk.inapp.images.memory
+
+import android.graphics.Bitmap
+import android.graphics.BitmapFactory
+import com.clevertap.android.sdk.inapp.images.hasValidBitmap
+import java.io.ByteArrayOutputStream
+import java.io.File
+
+enum class MemoryDataTransformationType {
+    MEMORY_DATA_TRANSFORM_TO_BITMAP,
+    MEMORY_DATA_TRANSFORM_TO_BYTEARRAY,
+    MEMORY_DATA_TRANSFORM_TO_FILE
+}
+
+val fileToBitmap: (file: File?) -> Bitmap? = { file ->
+    if (file != null && file.hasValidBitmap()) {
+        BitmapFactory.decodeFile(file.absolutePath)
+    } else {
+        null
+    }
+}
+val fileToBytes: (file: File?) -> ByteArray? = { file ->
+    file?.readBytes()
+}
+val bytesToBitmap: (bytes: ByteArray) -> Bitmap? = {
+    BitmapFactory.decodeByteArray(
+        it,
+        0,
+        it.size
+    )
+}
+val bitmapToBytes: (bitmap: Bitmap?) -> ByteArray? = {
+    it?.let {
+        val stream = ByteArrayOutputStream()
+        it.compress(Bitmap.CompressFormat.PNG, 100, stream)
```

Comment:
what was the quality in past, do we keep it as 100?

Reply:
It's taken from Jan release code of image downloading.